airbyte-notification/src/main/java/io/airbyte/notification/SlackNotificationClient.java

```diff
@@ -0,0 +1,128 @@
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.commons.yaml.Yamls;
+import io.airbyte.config.Notification;
+import io.airbyte.config.Notification.NotificationType;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.apache.logging.log4j.util.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Notification client that uses Slack API for Incoming Webhook to send messages.
+ *
+ * This class also reads a resource YAML file that defines the template message to send.
+ *
+ * It is stored as a YAML so that we can easily change the structure of the JSON data expected by
+ * the API that we are posting to (and we can write multi-line strings more easily).
+ *
+ * For example, slack API expects some text message in the { "text" : "Hello World" } field...
+ *
+ * It is also a template as the content of the message can be customized with basic variables that
+ * are built in the metadata map (re-using the same as generated by the JobTracker class for now)
+ * The templating is very basic for now: it is just looking for keys from the metadata map
+ * surrounded by '<' '>' characters in the string
+ */
+public class SlackNotificationClient implements NotificationClient {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SlackNotificationClient.class);
+
+  private final HttpClient httpClient = HttpClient.newBuilder()
+      .version(HttpClient.Version.HTTP_2)
+      .build();
+  private final Notification config;
+
+  public SlackNotificationClient(final Notification config) {
+    this.config = config;
+  }
+
+  @Override
+  public boolean notifyJobFailure(final Map<String, Object> jobData) throws IOException, InterruptedException {
+    if (config.getOnFailure()) {
+      return notify(renderJobData(jobData, "failure_slack_notification_template.yml"));
+    }
+    return true;
+  }
+
+  private static String renderJobData(final Map<String, Object> jobData, final String templateFile) throws IOException {
+    final Map<String, Object> metadata = MoreMaps.merge(jobData, generateMetadata(jobData));
+    final JsonNode template = Yamls.deserialize(MoreResources.readResource(templateFile));
+    String data = Jsons.toPrettyString(template);
+    for (Entry<String, Object> entry : metadata.entrySet()) {
+      final String key = "<" + entry.getKey() + ">";
+      if (data.contains(key)) {
+        data = data.replaceAll(key, entry.getValue().toString());
+      }
+    }
+    return data;
+  }
+
+  private static Map<String, Object> generateMetadata(final Map<String, Object> jobData) {
+    final Builder<String, Object> metadata = ImmutableMap.builder();
+    // TODO: Get the correct url to this airbyte instance
+    metadata.put("log_url", "http://localhost:8000/source/connection/" + jobData.get("connection_id"));
```

Comment:
Instead of hardcoding here, pass it in the constructor and hardcode in the consumer of this class